ledger/simulation/tracer.go

```diff
@@ -141,6 +143,13 @@ func (tracer *evalTracer) BeforeTxnGroup(ep *logic.EvalParams) {
 	if ep.FeeCredit != nil {
 		tracer.result.TxnGroups[0].FeeCredit = *ep.FeeCredit
 	}
+
+	// Override runtime related constraints against ep, before entering txn group
+	if tracer.config.UnLimitLog {
+		localDefaults := localConfig.GetDefaultLocal()
+		ep.MaxLogSize = localDefaults.SimulateLogBytesLimit
+		ep.MaxLogCalls = ep.MaxLogSize
```

Comment:
2048 feels like a long way from unlimited.

Reply:
Thanks for notifying me!  I was unaware of the relation between `config.MaxLogCalls` and `allocbound` on `EvalDelta` size, and I don't think it would bother raising the bar for `config.MaxLogCalls`, for it would affect consensus behavior, if I understood correctly?

If so, then I already let implementation fell back to using `config.MaxLogCalls`, though it won't give you way higher limit in log size.

Reply:
`config.MaxLogCalls` is not a consensus param, so we have some freedom with it. And note that this only controls the number of logs; the actual number of bytes from all the logs (`ep.MaxLogSize`) is not limited by this mechanism.

For `ep.MaxLogCalls`, an increase from 32 to 2048 seems pretty good. For `ep.MaxLogSize`, what if we make it max uint64? Since each log item can only be max 4k long and we already bound their number, this might be ok.

Reply:
Well 2048 * 4096 is a max of ~ 8MB of logs per app, which is a bit large. What about setting `ep.MaxLogSize` to 512000? This still allows apps to log the maximum-length 4k byte string 125 times, which seems plenty.

Reply:
ok, I get it.  I think I am on board with the 512000 size bound, and I understand that the log call bound should probably retain with `config.MaxLogCalls`, which is rather sufficient, I think:

the line is here https://github.com/algorand/go-algorand/blob/master/data/transactions/teal.go#L43

Then I propose we settle on:
1. log call limit derive from `config.MaxLogCalls`
2. log byte size limit stay as a config parameter

In this way, the allocbound should not be clobbered, and we still hold a large enough size of logging, sounds fine?

Reply:
see [remove local config simulation field](https://github.com/algorand/go-algorand/pull/5247/commits/5282967355cf30bc689b45b7bb87cf8071738b69) for latest relevant change on configuring `SimulateLogByteSize`.